src/lambda_codebase/initial_commit/bootstrap_repository/adf-bootstrap/deployment/example-global-iam.yml

```diff
@@ -23,4 +23,40 @@ Resources:
             Resource:
               - "*"
       Roles:
-        - adf-cloudformation-deployment-role
\ No newline at end of file
+        - adf-cloudformation-deployment-role
+
+  # ADFTerraformRole:
+  #   # This is the role that will be used to deploy Terraform resources from within 
+  #   # the target account. You should scope this policy depending on what you would 
+  #   # like to deploy within certain Organizational Units.
+  #   # NOTE: below is a sample IAM policy. This policies should NOT be used for purposes other than testing.
+  #   # Uncomment the `ADFTerraformRole` and `ADFTerraformPolicy` resources if you want to enable Terraform support in ADF.
+  #   Type: AWS::IAM::Role
+  #   Properties:
+  #     RoleName: "adf-terraform-role"
+  #     AssumeRolePolicyDocument:
+  #       Version: "2012-10-17"
+  #       Statement:
+  #         - Effect: Allow
+  #           Sid: "AssumeRole"
+  #           Principal:
+  #             AWS:
+  #               - !Sub arn:aws:iam::${AWS::AccountId}:role/adf-codebuild-role
```

Comment:
Instead of allowing the `adf-codebuild-role` to assume into this role, lets make it a separate role instead that is only assumed by CodeBuild itself when it needs to run Terraform apply.

All other stages can use the default `adf-codebuild-role` instead.